system/metrics/sinks/prometheus/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+################################################################################
+#    HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems®.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+################################################################################
+
+project(hpccmetrics_prometheus)
+
+set ( srcs
+        prometheusSink.cpp
+        prometheusReporterService.cpp
+        )
+
+include_directories(
+        ${HPCC_SOURCE_DIR}/system/include
+        ${HPCC_SOURCE_DIR}/system/jlib
+
+        ${HPCC_SOURCE_DIR}/system/security/shared
+        ${HPCC_SOURCE_DIR}/system/security/securesocket/
+        ${HPCC_SOURCE_DIR}/dali/base
+        ${HPCC_SOURCE_DIR}/system/mp
+
+        ${CMAKE_BINARY_DIR}/generated
+        ${HPCC_SOURCE_DIR}/esp/bindings/http/platform # http processing
+        ${HPCC_SOURCE_DIR}/esp/platform
+        ${HPCC_SOURCE_DIR}/esp/bindings
+        ${HPCC_SOURCE_DIR}/esp/clients
+        ${HPCC_SOURCE_DIR}/esp/bindings/SOAP/xpp
+        ${HPCC_SOURCE_DIR}/system/xmllib
+)
+
+ADD_DEFINITIONS(  -DPROMETHEUSSINK_EXPORTS )
+HPCC_ADD_LIBRARY( hpccmetrics_prometheussink SHARED ${srcs} )
+TARGET_LINK_LIBRARIES( hpccmetrics_prometheussink
+                       jlib
+                       esphttp
```

Comment:
I think esphttp is pulling in much too much code:

   ${XALAN_LIBRARIES}
    ${XERCES_LIBRARIES}
    jlib
    xmllib
    dalibase
    thorhelper      # this should not be dependent on the engine helper dll
    workunit        # neither should it be dependent on the wokunit dll
 
I think it will need a lighter weight library to process the requests.  @afishbeck any suggestions?

Reply:
Note, dafilesrv is one of the components that is likely to have metrics added.  It should be lightweight, and it would not pull in workunit, xmllib or xalan/xerces.

Reply:
Bringing in esphttp will include some of the base ESP metrics such as active request count. That would mean all components would get an ESP related active request count as part of their metrics as the metric is created statically.